dali/operators/reader/webdataset_reader_op.cc

```diff
@@ -0,0 +1,114 @@
+#include <algorithm>
+#include <cstring>
+#include <string>
+
+namespace dali {
+
+void WebdatasetReader::RunImpl(HostWorkspace& ws) {
+  int num_outputs = ws.NumOutput();
+  int num_samples = GetCurrBatchSize();
+
+  for (int data_idx = 0; data_idx < num_samples; data_idx++) {
+    auto& sample = GetSample(data_idx);
+    for (int output_idx = 0; output_idx < num_outputs; output_idx++) {
+      ws.OutputRef<CPUBackend>(output_idx)[data_idx].Resize(sample[output_idx].shape(),
+                                                            sample[output_idx].type());
+      ws.OutputRef<CPUBackend>(output_idx)[data_idx].SetMeta(sample[output_idx].GetMeta());
+      std::memcpy(ws.OutputRef<CPUBackend>(output_idx)[data_idx].raw_mutable_data(),
+                  sample[output_idx].raw_data(), sample[output_idx].nbytes());
+    }
+  }
+}
+
+DALI_SCHEMA(readers__Webdataset)
+    .DocStr((std::string) R"code(A reader for the webdataset format.
+
+The webdataset format is a way of providing efficient access to datasets stored in tar archives.
+
+Storing data in POSIX tar archives greatly speeds up I/O operations on mechanical storage devices 
+and on network file systems because it allows the operating system to reduce the number of I/O 
+operations and to read the data ahead.
+
+WebDataset fulfils a similar function to Tensorflow's TFRecord/tf.Example classes, but is much 
+easier to adopt because it does not actually require any data conversion. The data is stored in 
+exactly the same format inside tar files as it is on disk, and all preprocessing and data 
+augmentation code remains unchanged.
+
+The dataset consists of one or more tar archives, each of which is further split into samples. 
+A sample contains one or more components that correspond to the actual files contained within 
+the archive. The components that belong to a specific sample are aggregated by filename without 
+extension (for the specifics about the extensions please read the description of the ``ext`` parameter
+below). Note that samples with their filename starting with a dot will not be loaded, as well as 
+entries that are not regular files.
+
+In addition to the tar archive with data, each archive should come with a corresponding index file.
+The index file can be generated using a dedicated script as follows::
+
+    ``<path_to_dali>/tools/wds2idx.py <path_to_archive> <path_to_index_file>``
+
+The format of the index file is as follows:
+)code" + detail::wds::kCurrentIndexVersion + R"code( <num_samples>
+<component1_ext> <component1_offset> <component1_size> <component2_ext> <component2_offset> <component2_size> ...
+...
+
+
+Based on https://github.com/webdataset/webdataset)code")
+    .NumInput(0)
+    .OutputFn([](const OpSpec& spec) {
+      return spec.HasArgument("ext") ? spec.GetRepeatedArgument<std::string>("ext").size() : 0;
+    })
+    .AddArg("uris", R"code(The list of (one or more) paths to the webdataset archives.
```

Comment:
I don't think we use `uris` in other readers, how about just `path` ?

Reply:
alright